ports/stm/supervisor/internal_flash.c

```diff
@@ -95,12 +95,8 @@ STATIC const flash_layout_t flash_layout[] = {
 STATIC uint8_t _flash_cache[0x20000] __attribute__((aligned(4)));
 
 #elif defined(STM32L4)
-// todo - the L4 devices can have different flash sizes and different page sizes
-// depending upon the dual bank configuration
-// This is hardcoded for the Swan R5. When support for other devices is needed more conditionals will be required
-// to differentiate.
 STATIC const flash_layout_t flash_layout[] = {
-    { 0x08000000, 0x1000, 256 },
+    { 0x08100000, 0x1000, 256 },
```

Comment:
Want to take this value from the linker script? It is hard coded in internal_flash.h too. Centralizing to the linker would simplify things.

Reply:
I definitely concur with the intent.  However, I'm not sure that's possible, since the literal value is needed before link time.   We could add some code to populate the array at startup, but that feels like going one step too far for a relatively small concern.

Given that it's a well known value and hard-coded in the hardware, it's not something that needs to be configurable/flexible - so change is unlikely. The filesystem offsets were duplicated for the original F4/F7 ports, so the situation here isn't any "worse" than it was. 

Not meaning to be resistant here - if there's an easy way to achieve it, I'll gladly add it, but I'm thinking that if the original port didn't do it, then it's probably not that straightforward. 🤷

Reply:
No worries. Just glad to think it over.